sdk/typescript/src/types/objects.ts

```diff
@@ -70,30 +76,122 @@ export const SuiMovePackage = object({
+  assign(SuiRawMovePackage, object({ dataType: literal('package') })),
+]);
+export type SuiRawData = Infer<typeof SuiRawData>;
 
 export const MIST_PER_SUI = BigInt(1000000000);
 
-export const SuiObject = object({
-  /** The meat of the object */
-  data: SuiData,
-  /** The owner of the object */
-  owner: ObjectOwner,
-  /** The digest of the transaction that created or last mutated this object */
-  previousTransaction: TransactionDigest,
+export const ObjectDigest = string();
+export type ObjectDigest = Infer<typeof ObjectDigest>;
+
+export const SuiObjectData = object({
+  objectId: ObjectId,
+  version: SequenceNumber,
+  digest: ObjectDigest,
+  /**
+   * Type of the object, default to be undefined unless SuiObjectDataOptions.showType is set to true
+   */
+  type: optional(string()),
+  /**
+   * Move object content or package content, default to be undefined unless SuiObjectDataOptions.showContent is set to true
+   */
+  content: optional(SuiParsedData),
+  /**
+   * Move object content or package content in BCS bytes, default to be undefined unless SuiObjectDataOptions.showBcs is set to true
+   */
+  bcs: optional(SuiRawData),
+  /**
+   * The owner of this object. Default to be undefined unless SuiObjectDataOptions.showOwner is set to true
+   */
+  owner: optional(ObjectOwner),
+  /**
+   * The digest of the transaction that created or last mutated this object.
+   * Default to be undefined unless SuiObjectDataOptions.showPreviousTransaction is set to true
+   */
+  previousTransaction: optional(TransactionDigest),
   /**
    * The amount of SUI we would rebate if this object gets deleted.
    * This number is re-calculated each time the object is mutated based on
    * the present storage gas price.
+   * Default to be undefined unless SuiObjectDataOptions.showStorageRebate is set to true
    */
-  storageRebate: number(),
-  reference: SuiObjectRef,
+  storageRebate: optional(number()),
+});
+export type SuiObjectData = Infer<typeof SuiObjectData>;
+
+/**
+ * Config for fetching object data
+ */
+export const SuiObjectDataOptions = object({
+  /* Whether to fetch the object type, default to be true */
+  showType: optional(boolean()),
+  /* Whether to fetch the object content, default to be false */
+  showContent: optional(boolean()),
+  /* Whether to fetch the object content in BCS bytes, default to be false */
+  showBcs: optional(boolean()),
+  /* Whether to fetch the object owner, default to be false */
+  showOwner: optional(boolean()),
+  /* Whether to fetch the previous transaction digest, default to be false */
+  showPreviousTransaction: optional(boolean()),
+  /* Whether to fetch the storage rebate, default to be false */
+  showStorageRebate: optional(boolean()),
 });
-export type SuiObject = Infer<typeof SuiObject>;
+export type SuiObjectDataOptions = Infer<typeof SuiObjectDataOptions>;
+/**
+ * Util functions for constructing content options
+ */
+export function getObjectContentOptions(
```

Comment:
I don't think this is a blocker, but I'd remove this and just expect the caller to define the pieces they care about. Seems like an abstraction we don't need, but we can also just do this as a cleanup later on.